lib/runtime/storage/trie.go

```diff
@@ -61,7 +61,8 @@ func (s *TrieState) Snapshot() *trie.Trie {
 func (s *TrieState) BeginStorageTransaction() {
 	s.lock.Lock()
 	defer s.lock.Unlock()
-	s.oldTrie = s.t.Snapshot()
+	s.oldTrie = s.t
```

Comment:
`s.oldTrie`  points to the same ref as `s.t`, if I'm not wrong, any modifications in `s.t` will modify the `s.oldTrie`

Reply:
it does, but in the following line `s.t` gets snapshotted, so `s.oldTrie` won't be modified if `s.t` is modified when this function completes. `s.oldTrie` is effectively a pointer to the root of the previous trie, and `s.t` becomes a pointer to the root of the working trie

Reply:
Got it! just `s.t` changes the ref, `s.oldTrie` keeps pointing to the old, nice! Why not just keep `s.oldTrie = s.t.Snapshot` as before?

Reply:
`s.t.Snapshot()` now returns the new trie to be modified, previously it returned the old trie to be saved and `s.t` would continue to be modified